module/LearnZF2Themes/src/LearnZF2Themes/Controller/IndexController.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+namespace LearnZF2Themes\Controller;
+
+use Zend\Mvc\Controller\AbstractActionController;
+use Zend\View\Model\ViewModel;
+
+/**
+ * @author Stanimir Dimitrov <stanimirdim92@gmail.com>
+ */
+class IndexController extends AbstractActionController
+{
+    /**
+     * @var array
+     */
+    private $config = [];
+
+    /**
+     * @var array
+     */
+    private $themesConfig = [];
+
+    /**
+     * @var \Zend\ServiceManager\ServiceManager
+     */
+    protected $serviceLocator = null;
+
+    /**
+     * @method __construct
+     *
+     * @param array                               $config
+     * @param array                               $themesConfig
+     * @param \Zend\ServiceManager\ServiceManager $serviceLocator
+     */
+    public function __construct(array $config = [], array $themesConfig = [], \Zend\ServiceManager\ServiceManager $serviceLocator)
+    {
+        $this->config = $config;
+        $this->themesConfig = $themesConfig;
+        $this->serviceLocator = $serviceLocator;
```

Comment:
while `serviceLocator` is needed when overwrite `Config` service, I prefer move the process in `indexAction` to service instead.